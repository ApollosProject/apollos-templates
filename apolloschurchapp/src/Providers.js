import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <MediaPlayerProvider>
          <AnalyticsProvider>
            <LiveProvider>
              <Providers
                themeInput={customTheme}
                iconInput={customIcons}
                {...props}
              />
            </LiveProvider>
          </AnalyticsProvider>
        </MediaPlayerProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
