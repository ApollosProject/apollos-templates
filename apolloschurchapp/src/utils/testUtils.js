/* eslint-disable import/prefer-default-export */
import renderer from 'react-test-renderer';
import wait from 'waait';

export const renderWithApolloData = async (component) => {
  const tree = renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};
