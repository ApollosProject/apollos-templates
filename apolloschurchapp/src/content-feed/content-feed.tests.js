import React from 'react';

import Providers from '../Providers';

import { renderWithApolloData } from '../utils/testUtils';
import GET_CONTENT_FEED from './getContentFeed';

import ContentFeed from './index';

describe('content feed query component', () => {
  it('renders a feedview after successful query', async () => {
    const mock = {
      request: {
        query: GET_CONTENT_FEED,
        variables: { itemId: 'ContentChannel:123' },
      },
      result: {
        data: {
          node: {
            __typename: 'ContentChannel',
            childContentItemsConnection: {
              pageInfo: {
                endCursor: 'blablalba',
                __typename: 'PaginationInfo',
              },
              __typename: 'ContentItemsConnection',
              edges: [
                {
                  __typename: 'ContentItemsConnectionEdge',
                  node: {
                    __typename: 'UniversalContentItem',
                    id: 'UniversalContentItem:d57994350b9d213866b24dea3a97433d',
                    coverImage: null,
                    parentChannel: {
                      id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                      name: 'Editorial',
                      iconName: 'text',
                      __typename: 'ContentChannel',
                    },
                    title: 'Mea Animal Aperiam Ornatus Eu',
                    hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                    summary: 'Bla bla bla',
                    theme: null,
                    isLiked: false,
                    likedCount: 0,
                    videos: [],
                    audios: [],
                  },
                },
                {
                  __typename: 'ContentItemsConnectionEdge',
                  node: {
                    __typename: 'UniversalContentItem',
                    id: 'UniversalContentItem:b36e55d803443431e96bb4b5068147ec',
                    coverImage: null,
                    parentChannel: {
                      id: 'ContentChannel:4f68015ba18662a7409d1219a4ce013e',
                      name: 'Editorial',
                      iconName: 'text',
                      __typename: 'ContentChannel',
                    },
                    title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                    hyphenatedTitle:
                      'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                    summary: 'Bla bla bla',
                    theme: null,
                    isLiked: false,
                    likedCount: 0,
                    videos: [],
                    audios: [],
                  },
                },
              ],
            },
          },
        },
      },
    };

    const navigation = {
      getParam: () => 'ContentChannel:123',
      navigate: jest.fn(),
    };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]}>
        <ContentFeed navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
