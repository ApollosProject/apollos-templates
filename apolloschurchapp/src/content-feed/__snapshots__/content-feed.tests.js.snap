// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content feed query component renders a feedview after successful query 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "100%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "__typename": "UniversalContentItem",
          "coverImage": null,
          "hyphenatedTitle": "Mea Animal Aperiam Ornatus Eu",
          "id": "UniversalContentItem:d57994350b9d213866b24dea3a97433d",
          "summary": "Bla bla bla",
          "theme": null,
          "title": "Mea Animal Aperiam Ornatus Eu",
        },
        Object {
          "__typename": "UniversalContentItem",
          "coverImage": null,
          "hyphenatedTitle": "Probo Senserit Id Mea, Ut Sed Malis Postea,",
          "id": "UniversalContentItem:b36e55d803443431e96bb4b5068147ec",
          "summary": "Bla bla bla",
          "theme": null,
          "title": "Probo Senserit Id Mea, Ut Sed Malis Postea,",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.7}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onPressItem={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            isLoading={false}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "marginHorizontal": 16,
                "marginVertical": 12,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#A5A5A5",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "width": "100%",
                    }
                  }
                />
                <BVLinearGradient
                  colors={
                    Array [
                      2147483648,
                      0,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 0,
                      "y": 1,
                    }
                  }
                  locations={
                    Array [
                      0,
                      1,
                    ]
                  }
                  overlayColor="#000000"
                  overlayType="gradient-top"
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 32,
                    "paddingHorizontal": 24,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Black",
                      "fontSize": 24,
                      "lineHeight": 27.6,
                    }
                  }
                >
                  Mea Animal Aperiam Ornatus Eu
                </Text>
                <Text
                  bold={false}
                  italic={false}
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Regular",
                      "fontSize": 16,
                      "lineHeight": 26,
                      "paddingTop": 16,
                    }
                  }
                >
                  Bla bla bla
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            isLoading={false}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "marginHorizontal": 16,
                "marginVertical": 12,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#A5A5A5",
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "width": "100%",
                    }
                  }
                />
                <BVLinearGradient
                  colors={
                    Array [
                      2147483648,
                      0,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 0,
                      "y": 1,
                    }
                  }
                  locations={
                    Array [
                      0,
                      1,
                    ]
                  }
                  overlayColor="#000000"
                  overlayType="gradient-top"
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 32,
                    "paddingHorizontal": 24,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Black",
                      "fontSize": 24,
                      "lineHeight": 27.6,
                    }
                  }
                >
                  Probo Senserit Id Mea, Ut Sed Malis Postea,
                </Text>
                <Text
                  bold={false}
                  italic={false}
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Regular",
                      "fontSize": 16,
                      "lineHeight": 26,
                      "paddingTop": 16,
                    }
                  }
                >
                  Bla bla bla
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</BVLinearGradient>
`;
